chain/subscriber.go

```diff
@@ -377,12 +393,12 @@ func (cs *Subscriber) sync(index types.ChainIndex) error {
 			return fmt.Errorf("failed to fetch updates: %w", err)
 		}
 
-		// aggregate updates into one chain update
-		cu := NewChainUpdate()
-
 		// lock the subscriber while we apply the updates
 		cs.mu.Lock()
 
+		// aggregate updates into one chain update
+		cu := NewChainUpdate(cs.knownStateElements)
```

Comment:
I would fetch the `knownStateElements` from the database here instead of keeping them in memory. It's hard to keep that state consistent and it seems like the in-memory field doesn't consider reverted state elements right now